auth/auth.go

```diff
@@ -42,7 +42,7 @@ func LoadUserManager(settings *evergreen.Settings) (gimlet.UserManager, evergree
 	if authConfig.Naive != nil {
 		return makeNaiveManager(authConfig.Naive)
 	}
-	if authConfig.Github != nil {
+	if authConfig.Github != nil && authConfig.Github.ClientId != "" && authConfig.Github.ClientSecret != "" && authConfig.Github.Organization != "" && len(authConfig.Github.Users) > 0 {
```

Comment:
Can we put these checks into a `Validate` command so we can get a more descriptive error than what we'd end up with [here](https://github.com/evergreen-ci/evergreen/pull/7447/files#diff-f5c9a39fac1719f2a73593994b931d8a874449548e8a3753e1be46e761e503eaR51)?

Reply:
We could I guess, but I really just slapped on this fix to make the smoke test stop erroring (because when the smoke app server starts up, the GitHub app fields are filled in, but not the other stuff). We should just get rid of this logic someday, this is just legacy auth left over from when MongoDB used to change auth systems a lot before it settled on Okta.

Reply:
Alright SGTM

Reply:
I made DEVPROD-4088 in case we ever want to get rid of our legacy auth.